### PR TITLE
fix(deps): use npm-audit-resolver for npm audits

### PR DESCRIPTION
**Pull request by IanKrieger**
npm audit claimed there was a fix for the CVE below, however after the command was ran, nothing was actually fixed and the vulnerability still remained. Replace regular `npm audit` with the audit-resolve dep, so we can ignore faulty fixes / blocking dependencies when needed.

```
word-wrap  *
Severity: moderate
word-wrap vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-j8xg-fqg3-53r7
fix available via `npm audit fix`
node_modules/word-wrap
  optionator  0.8.3 - 0.9.1
  Depends on vulnerable versions of word-wrap
  node_modules/escodegen/node_modules/optionator
  node_modules/optionator
```

### CODE REVIEW

File: .github/workflows/deploy-to-production.yml
Comment (IanKrieger) [inline .github/workflows/deploy-to-production.yml:29]: audit is already run before merge, as we can catch things before merge this felt a bit unnecessary.

File: package.json
Comment (tackley) [inline package.json:28]: `npm add --exact` please - renovate will just raise a PR to pin this dependency by removing the `^` straight away afterwards

File: audit-resolve.json
Comment (tackley) [inline audit-resolve.json:11]: Actually in this particular case you can likely resolve this issue by updating the component that relies on word-wrap (it now uses a patched fork) - see https://github.com/brave/brave-talk/pull/1073 which did this.

File: package.json
Comment (tackley) [inline package.json:48]: Can we add in a `check` command like we have on ads-serve and the admin dashboard?  In a later PR is fine.

File: .github/workflows/deploy-to-production.yml
Comment (tackley) [reply .github/workflows/deploy-to-production.yml:29]: I'd like to keep this in - or actually to replace it with `run check:audit`, so at least we can suppress reviewed false positives.

Agree it's a bit unnecessary, and somewhat annoying due to the high rate of false positives on `npm audit` largely caused by the fact the ReDOS issues are largely irrelevant to a client-side web app. But the security team here have requested this in the past, just in case a true positive ever arose, so let's keep it in.

**Review by tackley — CHANGES_REQUESTED** [review]

**Review by tackley — APPROVED** [review]

**Pull request merged**
